Lib/inspect.py

```diff
@@ -3285,20 +3286,21 @@ def _bind(self, args, kwargs, *, partial=False):
 
             else:
                 if param.kind == _POSITIONAL_ONLY:
-                    # This should never happen in case of a properly built
-                    # Signature object (but let's have this check here
-                    # to ensure correct behaviour just in case)
-                    raise TypeError('{arg!r} parameter is positional only, '
-                                    'but was passed as a keyword'. \
-                                    format(arg=param.name))
-
-                arguments[param_name] = arg_val
+                    # Restore the param in case there is a kwargs_param
+                    kwargs[param_name] = arg_val
```

Comment:
I don't quite like this part. Never mutate the object that you are iterating over. Are the any other ways?

Reply:
Agree with the principle, but `kwargs` isn't iterated here.

That said, I can definitely refactor the surrounding code to avoid this "pop, then possibly put back" pattern. (I figure in the first version I was aiming for a smaller diff.)